resolwe_bio/processes/import_data/seq_reads.yml

```diff
@@ -218,3 +218,218 @@
+      type: list:basic:file
+    - name: fastqc_archive2
+      label: Download FastQC archive (Downstream)
+      type: list:basic:file
+  run:
+    runtime: polyglot
+    language: bash
+    program: |
+      FW_FASTQ=()
+      FW_FASTQGZ=()
+      FW_FASTQC=()
+      FW_FASTQC_URL=()
+
+      RW_FASTQ=()
+      RW_FASTQGZ=()
+      RW_FASTQC=()
+      RW_FASTQC_URL=()
+
+      {% for read in src1 %}
+        GZ_FILE_PATH={{ read.file.file }}
+        cp ${GZ_FILE_PATH} .
+
+        GZ_FILE={{ read.file.file|basename }}
+        gzip -dk "${GZ_FILE}"
+
+        NAME="${GZ_FILE%\.fastq\.gz}"
+        FW_FASTQ+=("${NAME}.fastq")
+        FW_FASTQGZ+=("${NAME}.fastq.gz")
+        FW_FASTQC+=("${NAME}_fastqc.zip")
+        FW_FASTQC_URL+=("{\"file\":\"fastqc/${NAME}_fastqc/fastqc_report.html\",\"refs\":[\"fastqc/${NAME}_fastqc\"]}")
+      {% endfor %}
+
+      {% for read in src2 %}
+        GZ_FILE_PATH={{ read.file.file }}
+        cp ${GZ_FILE_PATH} .
+
+        GZ_FILE={{ read.file.file|basename }}
+        gzip -dk "${GZ_FILE}"
+
+        NAME="${GZ_FILE%\.fastq\.gz}"
+        RW_FASTQ+=("${NAME}.fastq")
+        RW_FASTQGZ+=("${NAME}.fastq.gz")
+        RW_FASTQC+=("${NAME}_fastqc.zip")
+        RW_FASTQC_URL+=("{\"file\":\"fastqc/${NAME}_fastqc/fastqc_report.html\",\"refs\":[\"fastqc/${NAME}_fastqc\"]}")
+      {% endfor %}
+
+      FASTQ=(*.fastq)
+
+      echo "Postprocessing FastQC..."
+      mkdir "fastqc" && fastqc "${FASTQ[@]}" --extract --outdir="fastqc" 2> stderr.txt
+      re-checkrc "Failed while processing with FastQC."
+      if [[ $(grep --text "Failed to process file" stderr.txt) != "" ]]
+      then
+        re-error "Failed while processing with FastQC."
+      fi
+
+      mv fastqc/*_fastqc.zip .
+
+      for name in "${FW_FASTQ[@]}"; do
+        READS_NAME_FW=`basename "$name" .fastq`
+        ENCODING=`parse_encoding_type.py "./fastqc/${READS_NAME_FW}_fastqc/fastqc_data.txt"`
+
+        if [ "${ENCODING}" == "Illumina 1.5" ] ||  [ "${ENCODING}" == "Illumina 1.3" ]; then
+
+          echo "Recoding input reads from Phred64 encoding to Phred33 encoding."
+          mv "${READS_NAME_FW}.fastq" input_reads.fastq
+          TrimmomaticSE input_reads.fastq reformated.fastq.gz TOPHRED33
+          mv reformated.fastq.gz "${READS_NAME_FW}.fastq.gz"
+
+        elif [ "${ENCODING}" != "Sanger / Illumina 1.9" ]; then
+          re-error "Only Sanger / Illumina 1.9 / llumina 1.5 / Illumina 1.3 encoding is supported."
+        fi
+      done
+
+      for name in "${RW_FASTQ[@]}"; do
+        READS_NAME_RW=`basename "$name" .fastq`
+        ENCODING=`parse_encoding_type.py "./fastqc/${READS_NAME_RW}_fastqc/fastqc_data.txt"`
+
+        if [ "${ENCODING}" == "Illumina 1.5" ] ||  [ "${ENCODING}" == "Illumina 1.3" ]; then
+
+          echo "Recoding input reads from Phred64 encoding to Phred33 encoding."
+          mv "${READS_NAME_RW}.fastq" input_reads.fastq
+          TrimmomaticSE input_reads.fastq reformated.fastq.gz TOPHRED33
```

Comment:
Calling Trimmomatic with this command is not working on this docker image. It is working with ```java -jar /opt/usadellab/trimmomatic/trimmomatic-0.36.jar SE```, we should probably change the way trimmomatic is installed in `resolwebio/rnaseq:2.1.0` docker image

Reply:
Do not use the full path to call `trimmomatic`. I created [a script](https://github.com/genialis/resolwe-bio/blob/master/resolwe_bio/docker_images/rnaseq/packages-manual/trimmomatic.sh#L13) `trimmomatich.sh`  which is available on `$PATH` and we should use that instead.

Reply:
Please provide me a code snippet so I can update this like. We also need to update `upload-fastq-single`, etc.

Reply:
In all three cases just replace `TrimmomaticSE` with `trimmomatic.sh SE`. The rest of the parameters remain the same.

Reply:
Does this still apply?

Reply:
No. It's correct as you have it now. https://github.com/genialis/resolwe-bio/blob/master/resolwe_bio/docker_images/rnaseq/packages-manual/trimmomatic.sh#L13